[inductor][ez] ExternChoice with maybe_append_choice

coconutruben opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #162017
* #161534
* #161350
* #161351
* #161349
* #161348
* #161347
* #161346
* #161345
* #161344
* #161343
* #161342
* #161341
* #161340
* #161339
* #161338
* __->__ #161336
* #161126
* #161125
* #161124
* #161123

# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

Differential Revision: [D81520578](https://our.internmc.facebook.com/intern/diff/D81520578)

coconutruben committed:
[inductor][ez] ExternChoice with maybe_append_choice

\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


\# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

\# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

\# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

jansel commented on torch/_inductor/select_algorithm.py:
```diff
@@ -1901,6 +1903,19 @@ def bind(
             self, input_nodes, layout, kwargs, has_out_variant=self.has_out_variant
         )
 
+    def maybe_append_choice(
```
Is this used or tested anywhere?  PRs should include usage/tests and not add unused code.

jansel commented on torch/_inductor/select_algorithm.py:
```diff
@@ -1145,9 +1145,11 @@ def render(self, template, kwargs, record_input_dependent_tracked_event=False):
-            fn.__name__: self.record_input_dependent_tracked_event()(fn)
-            if record_input_dependent_tracked_event
-            else fn
+            fn.__name__: (
```
?

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][ez] ExternChoice with maybe_append_choice"


# why

- make the API for ExternChoice the same as KernelTemplate
- make it possible to use the same retrieval point as templates

# what

- add a maybe_append_choice to ExternChoice that under the hood
  invokes self.bind

This pr does not actuate the new path, but just exposes it

# testing

```
python3 -bb -m pytest test/inductor/test_max_autotune.py
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520578).
